src/native/libs/System.Globalization.Native/pal_calendarData.m

```diff
@@ -0,0 +1,158 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <assert.h>
+#include "pal_icushim_internal.h"
+#include "pal_calendarData.h"
+#import <Foundation/Foundation.h>
+
+#if defined(TARGET_MACCATALYST) || defined(TARGET_IOS) || defined(TARGET_TVOS)
+
+/*
+Function:
+GetCalendarIdentifier
+
+Gets the associated NSCalendarIdentifier for the CalendarId.
+*/
+static NSString* GetCalendarIdentifier(CalendarId calendarId)
+{
+    NSString *calendarIdentifier = NSCalendarIdentifierGregorian;
+    switch (calendarId)
+    {
+        case JAPAN:
+            calendarIdentifier = NSCalendarIdentifierJapanese;
+            break;
+        case THAI:
+            calendarIdentifier = NSCalendarIdentifierBuddhist;
+            break;
+        case HEBREW:
+            calendarIdentifier = NSCalendarIdentifierHebrew;
+            break;
+        case PERSIAN:
+            calendarIdentifier = NSCalendarIdentifierPersian;
+            break;
+        case HIJRI:
+            calendarIdentifier = NSCalendarIdentifierIslamic;
+            break;
+        case UMALQURA:
+            calendarIdentifier = NSCalendarIdentifierIslamicUmmAlQura;
+            break;
+        case TAIWAN:
+            calendarIdentifier = NSCalendarIdentifierRepublicOfChina;
+            break;
+        default:
+            calendarIdentifier = NSCalendarIdentifierGregorian;
+            break;
+    }
+    return calendarIdentifier;
+}
+
+/*
+Function:
+GlobalizationNative_GetCalendarInfoNative
+
+Gets a single string of calendar information for a given locale, calendar, and calendar data type.
+with the requested value.
+*/
+const char* GlobalizationNative_GetCalendarInfoNative(const char* localeName, CalendarId calendarId, CalendarDataType dataType)
+{
+    NSString *locName = [NSString stringWithFormat:@"%s", localeName];
+    NSLocale *currentLocale = [[NSLocale alloc] initWithLocaleIdentifier:locName];
+
+    NSString *calendarIdentifier = GetCalendarIdentifier(calendarId);
+    NSCalendar *calendar = [[NSCalendar alloc] initWithCalendarIdentifier:calendarIdentifier];
+
+    NSDateFormatter *dateFormat = [[NSDateFormatter alloc] init];
+    dateFormat.locale = currentLocale;
+    dateFormat.calendar = calendar;
+
+    switch (dataType)
+    {
+        case CalendarData_NativeName:
+            return calendar ? strdup([[calendar calendarIdentifier] UTF8String]) : NULL;
+        case CalendarData_MonthDay:
+        {
+             NSString *formatString = [NSDateFormatter dateFormatFromTemplate:@"MMMMd" options:0 locale:currentLocale];
+             return formatString ? strdup([formatString UTF8String]) : NULL;
+        }
```

Comment:
Not sure if this would be better, but perhaps we can avoid performing extra calculations for the CalendarDataTypes that don't need certain information.

For example, `CalendarData_MonthDay` and `CalendarData_YearMonths` both only need `currentLocale`, so we don't need to perform this calculation
```
    NSString *calendarIdentifier = GetCalendarIdentifier(calendarId);
    NSCalendar *calendar = [[NSCalendar alloc] initWithCalendarIdentifier:calendarIdentifier];

    NSDateFormatter *dateFormat = [[NSDateFormatter alloc] init];
    dateFormat.locale = currentLocale;
    dateFormat.calendar = calendar;
```
Maybe those could be placed at the top below `currentLocale`.

Reply:
That would be better. Will update.